modules/core/src/smithy4s/http/RawErrorResponse.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.http
+
+case class RawErrorResponse(
+    code: Int,
+    headers: Map[CaseInsensitive, Seq[String]],
+    body: String,
+    failedDecodeAttempt: Option[FailedDecodeAttempt]
+) extends Throwable {
+  override def getMessage(): String = {
+    val baseMessage = s"status $code, headers: $headers, body:\n$body"
+    failedDecodeAttempt match {
+      case Some(attempt) =>
+        baseMessage +
+          s"""
+             |FailedDecodeAttempt:
+             |  discriminator: ${attempt.discriminator}
+             |  contractError: ${attempt.contractError.getMessage}
+               """.stripMargin
+      case None => baseMessage
+    }
+  }
+}
+
+case class FailedDecodeAttempt(
```

Comment:
let's make this a Throwable with an implemented [getMessage](https://github.com/disneystreaming/smithy4s/pull/1570/files#diff-932de23e3329370510b6ff51fe738be5afca381f0c2ca081c0a3de894f38ba96R31-R33). Then in `RawErrorResponse`, you can have `def getCause(): Throwable = failedDecodeAttempt.orNull`

Reply:
Also @ghostbuster91, @GaryAghedo, let's take care of this requirement here: https://github.com/disneystreaming/smithy4s/pull/1575#issuecomment-2296559749